components/supervisor/pkg/supervisor/tasks.go

```diff
@@ -377,9 +379,29 @@ func (tm *tasksManager) watch(task *task, terminal *terminal.Term) {
 	go func() {
 		defer stdout.Close()
 
-		fileName := tm.prebuildLogFileName(task)
-		// TODO(janx): If the file already exists (from a parent prebuild), extract its "time saved", and log that below
-		// (instead, or in addition to, the incremental prebuild time).
+		var (
+			fileName                    = tm.prebuildLogFileName(task)
+			parentLog                   = ""
+			parentElapsed time.Duration = 0
+		)
+		if _, err := os.Stat(fileName); err == nil {
+			// If the file already exists (from a parent prebuild), extract its "elapsed time" before overwriting the prebuild logs.
+			content, err := ioutil.ReadFile(fileName)
```

Comment:
These log files can be arbitrarily large. Using a [`bufio.Scanner`](https://golang.org/pkg/bufio/#Scanner) to read the file line by line would be more appropriate (see [example](https://stackoverflow.com/questions/8757389/reading-a-file-line-by-line-in-go/16615559#16615559)).

Reply:
Good point, thanks! But my goal was also to repeat the previous prebuild logs (e.g. in case incremental prebuilds cause a subtle build problem, where you need to see the old logs + the new logs).

Maybe scan & write to a new file at the same time?

Reply:
You'd have to write the log to a temp file then, as otherwise you'd be overwriting yourself :)

Reply:
I've elected to move a pre-existing file to `{,-old}`, then read from that while writing / looking for the end message.